syncd/syncd_notifications.cpp

```diff
@@ -334,10 +334,29 @@ void process_on_fdb_event(
 
     bool sendntf = true;
 
+    std::vector<sai_fdb_event_notification_data_t> ntf_data;
+
+    sai_object_meta_key_t meta_key;
+    meta_key.objecttype = SAI_OBJECT_TYPE_SWITCH;
+    meta_key.objectkey.key.object_id = data[0].fdb_entry.switch_id;
+    sai_attribute_t attr;
+    attr.id = SAI_SWITCH_ATTR_SRC_MAC_ADDRESS;
+    auto info = sai_metadata_get_object_type_info(SAI_OBJECT_TYPE_SWITCH);
```

Comment:
please don't do any SAI queries in this functions, this is processed in separate thread, use the switch mac from SaiSwitch objects which has switch mac queried at instantiating swtich

also your approach will do query every time new notification will come which is not necessary